### PR TITLE
Add F1 score and log loss

### PR DESCRIPTION
**Pull request by romanlutz**
Also attempted to add F1-score to the widget as a performance metric, but I don't think that was sufficient. @rihorn2 is there any particular place where I need to do something other than the ones that I adjusted already? Perhaps I'm doing something wrong.

@MiroDudik if you don't want it part of the widget at this point we can also hold off on that and just do the metrics changes.

### CODE REVIEW

File: fairlearn/metrics/_balanced_root_mean_squared_error.py [hunk 1]
```diff
@@ -41,6 +41,6 @@ def _balanced_root_mean_squared_error(y_true, y_pred, sample_weight=None):
         y_ta_s = y_ta[indices]
         y_pa_s = y_pa[indices]
         s_w_s = s_w[indices]
-        errs[i] = math.sqrt(skm.mean_squared_error(y_ta_s, y_pa_s, s_w_s))
+        errs[i] = math.sqrt(skm.mean_squared_error(y_ta_s, y_pa_s, sample_weight=s_w_s))
```
Comment (riedgar-ms) [inline fairlearn/metrics/_balanced_root_mean_squared_error.py:44]: We had a lurking bug?

File: visualization/FairnessDashboard/src/Localization/en.json [hunk 2]
```diff
@@ -137,8 +140,8 @@
         "r2Description": "The fraction of variance in the labels explained by the model.",
         "aucDescription": "The quality of the predictions, viewed as scores, in separating positive examples from negative examples.",
         "balancedRMSEDescription": "Positive and negative examples are reweighted to have equal total weight. Suitable if the underlying data is highly imbalanced.",
-        "balancedAccuracy": "Balanced accuracy",
         "balancedAccuracyDescription": "Positive and negative examples are reweighted to have equal total weight. Suitable if the underlying data is highly imbalanced."
+        "f1ScoreDescription": "F1-Score is the harmonic mean of precision and recall."
```
Comment (riedgar-ms) [inline visualization/FairnessDashboard/src/Localization/en.json:144]: Any idea why the colon is showing up in red in the GitHub diff?

File: fairlearn/metrics/_balanced_root_mean_squared_error.py (hunk 1 above)
Comment (romanlutz) [reply fairlearn/metrics/_balanced_root_mean_squared_error.py:44]: Not a bug, but sklearn is deprecating certain positional args and making them keyword only. I saw the warnings and fixed it right away. I'm sure you don't mind :-)

File: visualization/FairnessDashboard/src/Localization/en.json (hunk 2 above)
Comment (romanlutz) [reply visualization/FairnessDashboard/src/Localization/en.json:144]: I'm so glad you spotted this! The comma from the previous line was missing. Not sure why I don't see errors when building the widget, though....

File: fairlearn/metrics/_balanced_root_mean_squared_error.py (hunk 1 above)
Comment (riedgar-ms) [reply fairlearn/metrics/_balanced_root_mean_squared_error.py:44]: Not at all :-)

**Comment by romanlutz** [comment]
> You won't see the TS updates until you rebuild and publish the `npm` package.

Are there steps for this?

File: fairlearn/widget/_fairlearn_dashboard.py [hunk 3]
```diff
@@ -129,6 +131,14 @@ def __init__(
                 "model_type": ["regression"],
                 "function": r2_score_group_summary
             },
+            "f1_score": {
+                "model_type": ["classification"],
+                "function": f1_score_group_summary
+            },
+            "log_loss": {
+                "model_type": ["regression"],
```
Comment (MiroDudik) [inline fairlearn/widget/_fairlearn_dashboard.py:139]: it should be `["probability"]`

File: visualization/FairnessDashboard/src/AccuracyMetrics.ts [hunk 4]
```diff
@@ -181,5 +181,20 @@ export const AccuracyOptions: {[key: string]: IAccuracyOption} = {
...
+        title: localization.Metrics.f1Score,
+        description: localization.Metrics.f1ScoreDescription,
+        isMinimization: false,
+        isPercentage: false,
+        userVisible: true
+    },
+    "log_loss": {
+        key: "log_loss",
+        title: localization.Metrics.logLoss,
+        description: localization.loremIpsum,
```
Comment (rihorn2) [inline visualization/FairnessDashboard/src/AccuracyMetrics.ts:196]: @romanlutz  or @MiroDudik  we definitely don't want placeholder description. Though this is currently set to not be user visible (is that intentional?)

Comment (romanlutz) [reply visualization/FairnessDashboard/src/AccuracyMetrics.ts:196]: I just copied from other metrics here. I was similarly surprised to see this (hence my message to you yesterday)

File: fairlearn/widget/_fairlearn_dashboard.py (hunk 3 above)
Comment (romanlutz) [reply fairlearn/widget/_fairlearn_dashboard.py:139]: so in _group_metric_set.py does it go under
BINARY_CLASSIFICATION_METRICS or
REGRESSION_METRICS?

I had it under regression so far.

File: visualization/FairnessDashboard/src/AccuracyMetrics.ts (hunk 4 above)
Comment (romanlutz) [reply visualization/FairnessDashboard/src/AccuracyMetrics.ts:200]: This isn't currently visible. If we want the ones currently with loremIpsum to be visible we need to write all the descriptions. Not doing that in this PR

File: visualization/FairnessDashboard/src/AccuracyMetrics.ts [hunk 5]
```diff
@@ -156,15 +157,17 @@ export const AccuracyOptions: {[key: string]: IAccuracyOption} = {
...
     },
     "root_mean_squared_error": {
         key: "root_mean_squared_error",
         title: localization.Metrics.rms_error,
         description: localization.Metrics.rmseDescription,
         isMinimization: true,
         isPercentage: false,
-        userVisible: true
+        userVisible: true,
+        alwaysUpperCase: true
```
Comment (rihorn2) [inline visualization/FairnessDashboard/src/AccuracyMetrics.ts:170]: great catch

File: visualization/FairnessDashboard/src/Controls/ModelComparisonChart.tsx [hunk 6]
```diff
@@ -182,9 +182,11 @@ export class ModelComparisonChart extends React.PureComponent<IModelComparisonPr
             return series;
         });
         const accuracyMetricTitle = selectedMetric.title;
```
Comment (rihorn2) [inline visualization/FairnessDashboard/src/Controls/ModelComparisonChart.tsx:184]: Github may be cutting out instances when I search, but it seems there are only two uses of this, both with the same ternary to do lowercase, so you could just set here using the ternary to correctly capitalize, then use the variable below without modification?

Comment (romanlutz) [reply visualization/FairnessDashboard/src/Controls/ModelComparisonChart.tsx:184]: done

File: fairlearn/widget/_fairlearn_dashboard.py (hunk 3 above)
Comment (MiroDudik) [reply fairlearn/widget/_fairlearn_dashboard.py]: this seems like a bug to me--we should have three task types, but let's leave that for a future PR. (tagging @riedgar-ms )

**Pull request merged**